Data/models/Domain.swift

```diff
@@ -85,60 +79,64 @@ public final class Domain: NSManagedObject, CRUD {
         return all(where: predicate, sortDescriptors: sortDescriptors) ?? []
     }
 
-    class func setBraveShield(forDomain domainString: String, state: Bool?, context: NSManagedObjectContext) {
-        // BRAVE TODO:
-//        guard let url = URL(string: domainString) else { return }
-//        let domain = Domain.getOrCreateForUrl(url, context: context)
-//        let shield = state.0
-//        switch (shield) {
-//            case .AllOff: domain?.shield_allOff = state.1 as NSNumber?
-//            case .AdblockAndTp: domain?.shield_adblockAndTp = state.1 as NSNumber?
-//            case .HTTPSE: domain?.shield_httpse = state.1 as NSNumber?
-//            case .SafeBrowsing: domain?.shield_safeBrowsing = state.1 as NSNumber?
-//            case .FpProtection: domain?.shield_fpProtection = state.1 as NSNumber?
-//            case .NoScript: domain?.shield_noScript = state.1 as NSNumber?
-//        }
-//        DataController.save(context: context)
+    public class func setBraveShield(forUrl url: URL, state: (BraveShieldState.Shield, Bool)) {
+        let context = DataController.newBackgroundContext()
+        
+        let domain = Domain.getOrCreateForUrl(url, context: context)
+        let (shield, setting) = (state.0, state.1 as NSNumber)
+        switch shield {
+            case .AllOff: domain?.shield_allOff = setting
+            case .AdblockAndTp: domain?.shield_adblockAndTp = setting
+            case .HTTPSE: domain?.shield_httpse = setting
+            case .SafeBrowsing: domain?.shield_safeBrowsing = setting
+            case .FpProtection: domain?.shield_fpProtection = setting
+            case .NoScript: domain?.shield_noScript = setting
+        }
+        
+        DataController.save(context: context)
+        
+        // After save update app state
+        BraveShieldState.set(forUrl: url, state: state)
     }
 
-    class func loadShieldsIntoMemory(_ completionOnMain: @escaping () -> Void) {
-        // Brave TODO:
-//        BraveShieldState.perNormalizedDomain.removeAll()
+    // TODO: Move logic to ShieldState init
+    public class func loadShieldsIntoMemory(_ completionOnMain: @escaping () -> Void) {
```

Comment:
Only place I see this used doesn't use the completion block. Should it be optional?